types/analysis.ts

```diff
@@ -8,8 +8,9 @@ export interface Ranking {
 
 export interface Analysis {
   colors: string[];
-  description: string
-  ranking: Ranking;
+  description: string;
+  // ToDo: deprecate Ranking
+  ranking?: Ranking;
```

Comment:
There sometime will and will not be ranking. I'm going to make a separate pull request to remove the ranking stuff before I continue typing to make things easier.